### DISCUSSION TITLE
Optimal Transport Distance value in domain adaptation

### DISCUSSION THREAD

**Discussion by XLoizeau** [PythonOT/POT#645] [Q&A] [answered]
Hello,
I am currently working on a software that relies on the computation of OT distance between colour palettes.

Basing myself on this [example from the official gallery](https://pythonot.github.io/auto_examples/domain-adaptation/plot_otda_mapping_colors_images.html#sphx-glr-auto-examples-domain-adaptation-plot-otda-mapping-colors-images-py) I was a bit surprised to see that the distance is not returned plainly.

After a bit of digging around, I saw that when using the Earth Movers distance as follows:
```python
ot_emd = ot.da.EMDTransport()
ot_emd.fit(Xs = Xs, Xt = Xt)
transp_Xs_emd = ot_emd.transform(Xs = X1)
```
the distance seems to be stored in the log_ attribute of `ot_emd` in the `cost` key. Yet, I am not entirely sure as the documentation about those attributes is a bit sibylline.

When it comes to the other distances (sinkhorn and mapping transport) demonstrated in this example, I could not find the distance at all.

Would anyone be able to tell me if I am missing something and if there is a reason for those quantities not to be returned?

Thank you so much for your help.
Best,
Zhav

**Comment by rflamary** [comment, marked as answer]
The OT distance is returned by function [ot.emd2](https://pythonot.github.io/all.html#ot.emd2) same with sinkhorn

```python
# a,b are 1D histograms (sum to 1 and positive)
# M is the ground cost matrix
Wd = ot.emd2(a, b, M) # exact linear program
Wd_reg = ot.sinkhorn2(a, b, M, reg) # entropic regularized OT
# if b is a matrix compute all distances to a and return a vector
```

Or we have the more recent [`ot.solve`](https://pythonot.github.io/all.html#ot.solve) function that returns a class containing the OT plan `ot.solve(...).plan` ad the value `ot.solve(...).value`

**Reply by XLoizeau** [reply]
Oh... I feel a bit stupid now... Thank you for taking the time to answer!
Best,
Zhav
